### PR TITLE
adding random seed for gen_trace to encoder testing

### PR DESCRIPTION
**Pull request by crysmags**
## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: tests/benchmarks/test_encoding.py [hunk 1]
```diff
@@ -24,31 +26,25 @@ def decode(data):
         return msgpack.unpackb(data, raw=True)
 
 
-trace_large = gen_trace(nspans=1000)
+# Set a static seed to ensure gen_trace is consistant
+random.seed(13)
```
Comment (brettlangdon) [inline tests/benchmarks/test_encoding.py:30]: @P403n1x87 is this the right place for this? should we put it in `gen_trace` instead or maybe a `conftest.py` ?

**Review by brettlangdon — COMMENTED** [review]
one question for others about where we should put `random.seed` but otherwise lgtm!

File: tests/benchmarks/test_encoding.py [hunk 2]
```diff
@@ -24,31 +26,25 @@ def decode(data):
...
 
-@pytest.mark.benchmark(group="encoding.join_encoded", min_time=0.005)
-def test_join_encoded(benchmark):
-    benchmark(
-        msgpack_encoder.join_encoded,
-        [msgpack_encoder.encode_trace(trace_large), msgpack_encoder.encode_trace(trace_small)],
-    )
-
-
+# group name: encoding
```
Comment (Kyle-Verhoog) [inline tests/benchmarks/test_encoding.py:39]: Is this comment necessary?

File: tests/benchmarks/test_encoding.py [hunk 3]
```diff
@@ -57,23 +53,84 @@ def test_encode_1000_span_trace_fallback(benchmark):
...
 def test_encode_1000_span_trace_custom(benchmark):
     benchmark(trace_encoder.encode_trace, trace_large)
 
 
+@pytest.mark.benchmark(group="encoding", min_time=0.005)
+def test_encode_500_span_trace_custom(benchmark):
+    benchmark(trace_encoder.encode_trace, trace_medium)
+
+
+# group name: encoding.small
```
Comment (Kyle-Verhoog) [inline tests/benchmarks/test_encoding.py:72]: Same here?

File: tests/benchmarks/test_encoding.py [hunk 4]
```diff
@@ -57,23 +53,84 @@ def test_encode_1000_span_trace_fallback(benchmark):
...
+
+
 @pytest.mark.benchmark(group="encoding.join_encoded", min_time=0.005)
 def test_join_encoded_custom(benchmark):
     benchmark(
-        trace_encoder.join_encoded, [trace_encoder.encode_trace(trace_large), trace_encoder.encode_trace(trace_small)]
+        trace_encoder.join_encoded,
+        [
+            trace_encoder.encode_trace(trace_large),
+            msgpack_encoder.encode_trace(trace_medium),
```
Comment (Kyle-Verhoog) [inline tests/benchmarks/test_encoding.py:133]:
```suggestion
            trace_encoder.encode_trace(trace_medium),
```
?

**Review by Kyle-Verhoog — APPROVED** [review]
lgtm! Where we put the `seed()` call is an interesting question which I'm not sure I know the answer to but it's in our tests so we can easily update it later on if need be 🙂

File: tests/benchmarks/test_encoding.py (hunk 1 above)
Comment (P403n1x87) [reply tests/benchmarks/test_encoding.py]: If we want to control the seed for all tests then yes we should either put it in `test_encoders` or even better in a `conftest.py`. Perhaps we should also set `PYTHONHASHSEED`, e.g. `PYTHONHASHSEED=0` before running `pytest` to also control the hash seed.

**Review by jd — CHANGES_REQUESTED** [review]
I'd like to read the motivation for such a change. I'm sure it's for the better, but I've no clue why we're doing this reading the pull request body.

**Comment by P403n1x87** [comment]
> I'd like to read the motivation for such a change. I'm sure it's for the better, but I've no clue why we're doing this reading the pull request body.

This is an attempt to remove as much randomness as possible from the benchmark runs. In order for the results to be comparable we want to generate the same data and ensure that Python dicts all have the same chances of collisions etc.... I think in the future we'd want to move to a fixture-based approach with pre-generated synthetic data.

**Comment by jd** [comment]
@P403n1x87 all fine, but that should be somewhere in `git log`. :)

**Comment by majorgreys** [comment]
This work will be taken up in #2646

**Pull request closed without merging**